ndonnx/_core/_shapeimpl.py

```diff
@@ -24,6 +25,15 @@ def shape(self, x):
             current = next(iter(current._fields.values()))
         return from_corearray(opx.shape(current))
 
+    def static_shape(self, x) -> tuple[int | None, ...]:
+        current = x
+        while isinstance(current, ndx.Array):
```

Comment:
```suggestion
        # FIXME: Implicit assumption that all fields represent the shape of `x` correctly
        while isinstance(current, ndx.Array):
```

Reply:
We are in `_shapeimpl.py` which is used in dtypes that satisfy that condition - see https://github.com/Quantco/ndonnx/blob/main/ndonnx/_core/_shapeimpl.py#L17-L19.